config/v1/types_authentication.go

```diff
@@ -105,13 +107,17 @@ const (
 
 // webhookTokenAuthenticator holds the necessary configuration options for a remote token authenticator
 type WebhookTokenAuthenticator struct {
-	// kubeConfig contains kube config file data which describes how to access the remote webhook service.
+	// kubeConfig references a secret that contains kube config file data which
+	// describes how to access the remote webhook service.
+	// The namespace for the referenced secret is openshift-config.
+	//
 	// For further details, see:
+	//
 	// https://kubernetes.io/docs/reference/access-authn-authz/authentication/#webhook-token-authentication
+	//
 	// The key "kubeConfig" is used to locate the data.
 	// If the secret or expected key is not found, the webhook is not honored.
 	// If the specified kube config data is not valid, the webhook is not honored.
-	// The namespace for this secret is determined by the point of use.
 	KubeConfig SecretNameReference `json:"kubeConfig"`
```

Comment:
Is this required? I think it is if this struct is non-null. Add minLength:1 and make it required via tag.

See https://book.kubebuilder.io/reference/markers/crd-validation.html.

Reply:
webhook token authenticator is defined by the kubeconfig, so it is required. I added the `//+required` tag, although it might be the default.

Can `minLength` be used on structs?

Reply:
no, it can't. I assumed it is an alias of string. Then you have mark it as required. The name field inside is already required.

Reply:
Btw, the default is optional, not required. Compare the `// +kubebuilder:validation:Optional` in doc.go.